docs/internals/sorting.md

```diff
@@ -0,0 +1,93 @@
+# Sorting
+
+Quickwit can sort results based on fastfield values or score. This document discuss where and ho
+ it happens.
+It also tries to describe optimizations that may be enabled (but are not necessarily implemente)
+by this behavior.
+
+# Current state
+(this section shall be removed once proposed changes are accepted)
+
+## Behavior
+
+Sorting is controlled by the `sort_by` query parameter. It accepts a comma separated list of fields
+to use for sorting. Sorting is Ascending by default. The sorting order can be reversed by prefixing
+a field name with a hyphen `-`.
+The special value `_score` means sorting by score, with a sort that is then Descending by default
+(documents matching better goes first).
+
+In case of equality between two documents, the GlobalDocId, composed of (SplitId, SegmentId, DocId)
+is used as a tie breaker. It is always used in Ascending order *inside a split*. It is used in the
+same order as the first field *when merging results*
+// TODO is this really the case? That's at least what my own tests seems to show.
+
+If a document doesn't have a value for a sorting field, that document is considered to go after any
+document which has a value, independantly of sort order. That is, when sorting the value 1,2 and
+None, ascending sort would give `[1, 2, None]`, and descending sort would give `[2, 1, None]`.
+
+If the client does not request sorting, documents are sorted using (DocId, SplitId, SegmentId), in
+Descending order.
```

Comment:
I suspect you made an error.

```suggestion
If the client does not request sorting, documents are sorted using (SplitId, SegmentId, DocId), in
Descending order. In other words, everything happens as if documents were sorted by a constant value.
```

(This suggests that descending order by default is not a bad idea)

Reply:
as surprising as it seems, this is the actual order used. We use `(SplitId, SegmentId, DocId)` as a tie breaker, but sort primarily based on a [value which is the DocId](https://github.com/quickwit-oss/quickwit/blob/5d2c66e38e17ba5952196f01123d87f5a98b6a3c/quickwit/quickwit-search/src/collector.rs#L203). It might be more explicit to write that we sort by `(DocId, SplitId, SegmentId, DocId)`, which is a bit redundant, but more of what's actually happening